app/addons/compaction/components.react.jsx

```diff
@@ -0,0 +1,169 @@
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not
+// use this file except in compliance with the License. You may obtain a copy of
+// the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+// License for the specific language governing permissions and limitations under
+// the License.
+
+define([
+  'app',
+  'api',
+  'react',
+  'addons/compaction/stores',
+  'addons/compaction/actions',
+  'addons/components/react-components.react'
+],
+
+function (app, FauxtonAPI, React, Stores, Actions, Components, ReactComponents) {
+  var compactionStore = Stores.compactionStore;
+
+  var CompactDatabase = React.createClass({
+
+    run: function (e) {
+      this.props.compactDatabase();
+    },
+
+    render: function () {
+      var btnText = 'Run';
+
+      if (this.props.isCompacting) {
+        btnText = 'Compacting...';
+      }
+
+      return (
+        <div className="row-fluid">
+          <div className="span12 compaction-option">
+            <h3> Compact Database </h3>
```

Comment:
pls remove the sourrounding space.

it leads to the invariant issues we had with react